metadata/metadata.go

```diff
@@ -86,11 +93,14 @@ var parentMapping = map[ResourceType]ResourceType{
 
 func (serv *MetadataServer) needsJob(res Resource) bool {
 	if res.ID().Type == TRAINING_SET_VARIANT ||
-		res.ID().Type == FEATURE_VARIANT ||
 		res.ID().Type == SOURCE_VARIANT ||
 		res.ID().Type == LABEL_VARIANT {
 		return true
 	}
+	if res.ID().Type == FEATURE_VARIANT {
+		fv := res.(*featureVariantResource)
```

Comment:
I dont love that this will panic if for some reason its not a featureVariantResource. Would recommend either doing
`fv, ok := res.(*featureVariantResource)` and logging a nasty error then returning false, or adding an error to the return